genesyscloud/integration_action_draft/resource_genesyscloud_integration_action_draft.go

```diff
@@ -0,0 +1,206 @@
+	"github.com/mypurecloud/terraform-provider-genesyscloud/genesyscloud/util"
+	"github.com/mypurecloud/terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"log"
+	"strings"
+	"time"
+)
+
+func getAllIntegrationActionDrafts(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	iap := getIntegrationActionsProxy(clientConfig)
+
+	actions, resp, err := iap.getAllIntegrationActionDrafts(ctx, "")
+	if err != nil {
+		return nil, util.BuildAPIDiagnosticError(ResourceType, fmt.Sprintf("Failed to get integration action drafts %s", err), resp)
+	}
+
+	for _, action := range *actions {
+		// Don't include "static" actions
+		if strings.HasPrefix(*action.Id, "static") {
+			continue
+		}
+		resources[*action.Id] = &resourceExporter.ResourceMeta{BlockLabel: *action.Name}
+	}
+	return resources, nil
+}
+
+func createIntegrationActionDraft(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	iap := getIntegrationActionsProxy(sdkConfig)
+	name := d.Get("name").(string)
+
+	log.Printf("Creating integration action draft %s", name)
+
+	draftRequest := buildActionDraftFromResourceData(d)
+
+	draft, resp, err := iap.createIntegrationActionDraft(ctx, *draftRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(ResourceType, fmt.Sprintf("Failed to create integration action draft %s error: %s", name, err), resp)
+	}
+
+	d.SetId(*draft.Id)
+	log.Printf("Created integration action draft %s %s", name, *draft.Id)
+
+	return readIntegrationActionDraft(ctx, d, meta)
+}
+
+func readIntegrationActionDraft(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	iap := getIntegrationActionsProxy(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceIntegrationActionDraft(), constants.ConsistencyChecks(), ResourceType)
+
+	log.Printf("Reading integration action draft %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		draft, resp, getErr := iap.getIntegrationActionDraftById(ctx, d.Id())
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(ResourceType, fmt.Sprintf("Failed to read action draft %s: %s", d.Id(), getErr), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(ResourceType, fmt.Sprintf("Failed to read action draft %s: %s", d.Id(), getErr), resp))
+		}
+
+		// Retrieve config request/response templates
+		reqTemp, resp, err := iap.getIntegrationActionDraftTemplate(ctx, d.Id(), reqTemplateFileName)
+		if err != nil {
+			if util.IsStatus404(resp) {
+				d.SetId("")
+				return nil
```

Comment:
Why are we setting the id to nil here? And same below. Would be good to have some logs here also